src/darker/tests/test_import_sorting.py

```diff
@@ -8,3 +13,28 @@ def test_apply_isort():
     result = apply_isort(ORIGINAL_SOURCE)
 
     assert result == ISORTED_SOURCE
+
+
+@pytest.mark.parametrize("settings_file", [None, "pyproject.toml"])
+@pytest.mark.parametrize("line_length", [20, 60])
+def test_isort_config(monkeypatch, tmpdir, line_length, settings_file):
+    from black import find_project_root
+
+    find_project_root.cache_clear()
+    monkeypatch.chdir(tmpdir)
+    (tmpdir / 'pyproject.toml').write(
+        dedent(
+            f"""\
+            [tool.isort]
+            line_length = {line_length}
+            """
+        )
+    )
+
+    content = "from module import ab, cd, ef, gh, ij, kl, mn, op, qr, st, uv, wx, yz"
+    src = Path(tmpdir / "test1.py") if not settings_file else None
+    config = str(tmpdir / settings_file) if settings_file else None
+
+    actual = apply_isort(content, src, config)
+    expected = apply_isort(content, line_length=line_length)
+    assert actual == expected
```

Comment:
Ok, I see now where the need for `src: Optional[Path]` came from.

My suggestion would be to have explicit expected results instead of relying on calling the tested function twice with different arguments to verify its own output. If a bug is introduced in that function, it could very well be that both of those use cases are affected, and the test reveals nothing!

This also could make it more obvious to the reader of the code what actually is happening.

```suggestion
@pytest.mark.parametrize(
    "line_length, settings_file, expect",
    [
        (
            50,
            None,
            "from module import (ab, cd, ef, gh, ij, kl, mn,\n"
            "                    op, qr, st, uv, wx, yz)\n",
        ),
        (
            50,
            "pyproject.toml",
            "from module import (ab, cd, ef, gh, ij, kl, mn,\n"
            "                    op, qr, st, uv, wx, yz)\n",
        ),
        (
            60,
            None,
            "from module import (ab, cd, ef, gh, ij, kl, mn, op, qr, st,\n"
            "                    uv, wx, yz)\n",
        ),
        (
            60,
            "pyproject.toml",
            "from module import (ab, cd, ef, gh, ij, kl, mn, op, qr, st,\n"
            "                    uv, wx, yz)\n",
        ),
    ],
)
def test_isort_config(monkeypatch, tmpdir, line_length, settings_file, expect):
    find_project_root.cache_clear()
    monkeypatch.chdir(tmpdir)
    (tmpdir / 'pyproject.toml').write(
        dedent(
            f"""\
            [tool.isort]
            line_length = {line_length}
            """
        )
    )

    content = "from module import ab, cd, ef, gh, ij, kl, mn, op, qr, st, uv, wx, yz"
    src = Path(tmpdir / "test1.py") if not settings_file else None
    config = str(tmpdir / settings_file) if settings_file else None

    actual = apply_isort(content, src, config)
    assert actual == expect
```